### PR TITLE
feat: add ability to update exsting static tables

### PR DESCRIPTION
**Pull request by fontanierh**
## Description

fixes https://github.com/dust-tt/tasks/issues/308

builders will be able to modify a static table description and replace the CSV/TSV file with a new one.

## Risk

code changes have limited blast radius (structured data features only)

## Deploy Plan

Nothing special

### CODE REVIEW

File: front/pages/api/w/[wId]/data_sources/[name]/tables/csv.ts [hunk 1]
```diff
@@ -94,26 +100,28 @@ async function handler(
...
+      if (csvRowsRes?.isErr()) {
         return apiError(req, res, {
           api_error: csvRowsRes.error,
           status_code: 400,
         });
       }
 
-      const csvRows = csvRowsRes.value;
-      if (csvRows.length > 500_000) {
+      const csvRows = csvRowsRes?.unwrap();
```
Comment (fontanierh) [inline front/pages/api/w/[wId]/data_sources/[name]/tables/csv.ts:112]: needed the unwrap feature as checking `csvRowsRes?.isErr()` above doesn't do the type narrowing as one would hope :/

File: front/pages/w/[wId]/builder/data-sources/[name]/tables/upsert.tsx [hunk 2]
```diff
@@ -260,7 +299,10 @@ export default function TableUpsert({
                 name="table-name"
                 disabled={readOnly || !!loadTableId}
                 value={tableName}
-                onChange={(v) => setTableName(v)}
+                onChange={(v) => {
+                  setEdited(true);
```
Comment (PopDaph) [inline front/pages/w/[wId]/builder/data-sources/[name]/tables/upsert.tsx:303]: NIT: Means I can submit if I add and remove a char right?

**Review by PopDaph — APPROVED** [review]

File: front/pages/w/[wId]/builder/data-sources/[name]/tables/upsert.tsx [hunk 3]
```diff
@@ -110,16 +112,24 @@ export default function TableUpsert({
...
-  }, [tableName, description, file]);
+    if (!loadTableId && !file) {
+      // File can be null if we are editing a table.
+      return setDisabled(true);
+    }
+    if (!tableName || !description) {
+      return setDisabled(true);
+    }
+
+    return setDisabled(!edited);
```
[the remark this answers is not in the document]
Comment (fontanierh) [reply front/pages/w/[wId]/builder/data-sources/[name]/tables/upsert.tsx:123]: Fair, agree it's better 👍

File: front/pages/api/w/[wId]/data_sources/[name]/tables/csv.ts (hunk 1 above)
Comment (fontanierh) [reply front/pages/api/w/[wId]/data_sources/[name]/tables/csv.ts:112]: Yes this would work 👍

**Pull request merged**